### ISSUE TITLE
Unable to write new statefile in TFC if `tfm copy workspaces --state` is used to migrate states

### ISSUE THREAD

**Issue by drewmullen** [hashicorp-services/tfm#139]
**Describe the bug**
We're performing a migration from TFE version `v2021112-2` to TFC. Using the `tfm .. --state` command is failing to produce usable workspaces. By process of elimination, we have been able to isolate that the `--state` function is where the error breaks down

### What fails:

if we use `tfm` to migrate the statefiles the copy is successful but **subsequent applys** make the expected changes to our managed resources but then **fails to write the new statefile**. We are able to recreate this error locally by running TFC in `local` mode and then attempting to `terraform state push` the `errored.tfstate` file. You can see this exact error below:  

### What is successful:

If we use `tfm` to copy **only** the workspaces then use `terraform state push` to manually add the statefiles to TFC we're able to successfully perform normal operations (apply, etc).

ERROR from command `terraform state push` to the workspace migrated using `tfm copy workspaces --state`

```shell
│ Error: Failed to save state
│ 
│ Error saving state: error uploading state: 409 Conflict
╵
╷
│ Error: Failed to persist state to backend
│ 
│ The error shown above has prevented Terraform from writing the updated
│ state to the configured backend. To allow for recovery, the state has been
│ written to the file "errored.tfstate" in the current working directory.
│ 
│ Running "terraform apply" again at this point will create a forked state,
│ making it harder to recover.
│ 
│ To retry writing this state, use the following command:
│     terraform state push errored.tfstate
```

Notes:
- Plans work in tfc
- We can recreate the error in TFC, CLI, via CI/CD pipeline (that uses API calls)

**Comment by jeffmccollum** [comment]
@Josh-Tracy and I were able to figure this out. A new release will be out in the morning!

**Comment by jeffmccollum** [comment]
This should be fixed now in 0.4.1. Please let us know your results!
